### PR TITLE
feat(eks): support INF2 instance types

### PR DESCRIPTION
**Pull request by freschri**
fix: added INF2 support to 1/ isGpuInstanceType to correctly select AMI, 2/ neuron-device-plugin-daemonset

INF2 is currently (wrongly) not included in the list of instance types mapping to GPU AMIs.
The change adds it to the list

inf2 not present in neuron-device-plugin-daemonset, added

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by kaizencc — DISMISSED** [review]
are the sagemaker changes and the eks changes independent of each other? if so, I'd prefer them being included in two separate PRs. Otherwise, this looks largely ok. The integ test will need to be run, however.

File: packages/aws-cdk-lib/aws-eks/lib/addons/neuron-device-plugin.yaml [hunk 1]
```diff
@@ -40,7 +40,11 @@ spec:
                       - inf1.xlarge
                       - inf1.2xlarge
                       - inf1.6xlarge
-                      - inf1.4xlarge
```
[the remark this answers is not in the document]
Comment (freschri) [reply packages/aws-cdk-lib/aws-eks/lib/addons/neuron-device-plugin.yaml:43]: yes, see list here: https://aws.amazon.com/ec2/instance-types/inf1/

File: packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-inference.ts [hunk 2]
```diff
@@ -25,6 +25,11 @@ class EksClusterInferenceStack extends Stack {
       instanceType: new ec2.InstanceType('inf1.2xlarge'),
       minCapacity: 1,
     });
+    
+    cluster.addAutoScalingGroupCapacity('InferenceInstances', {
+      instanceType: new ec2.InstanceType('inf2.xlarge'),
+      minCapacity: 1,
+    });
```
[the remark this answers is not in the document]
Comment (freschri) [reply packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-inference.ts:32]: I made further changes: I duplicated the integ test: 1 with asg inf1, the other inf2.
It is in failed state currently, not sure if I need to actually do something manually somewhere:
aws-cdk-eks-cluster-inf1-test: destroy failed Error: The stack named aws-cdk-eks-cluster-inf1-test is in a failed state. You may need to delete it from the AWS console : DELETE_FAILED (The following resource(s) failed to delete: [ClusterNodegroupDefaultCapacityNodeGroupRole55953B04, ClusterInf1InstancesInstanceRole67C931E4]. )

[the remark this answers is not in the document]
Comment (kaizencc) [reply packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-inference.ts:32]: actually i can try to run this for you. our eks integ tests take forever and are wonky :(

**Review by kaizencc — APPROVED** [review]

**Pull request merged**